src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,497 @@
+
+To index multiple tables, you'll need to configure multiple pipelines in the configuration. For further guidance, please consult the [pipeline section](https://opensearch.org/docs/latest/data-prepper/pipelines/pipelines/) of the OpenSearch documentation.
+
+
+NOTE: An OpenSearch Ingestion pipeline supports only one DynamoDB table as its source. For more details on current limitations, Please refer to [Amazon OpenSearch Limitation](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/configure-client-ddb.html#ddb-pipeline-limitations) section.
+
+Now, create the OSIS pipeline resource:
+
+
+``` ts title="amplify/backend.ts"
+
+// Create a CloudWatch log group
+const logGroup = new logs.LogGroup(openSearchStack, "LogGroup", {
+  logGroupName: "/aws/vended-logs/OpenSearchService/pipelines/1",
+  removalPolicy: RemovalPolicy.DESTROY,
+});
+
+// Create an OpenSearch Integration Service pipeline
+const cfnPipeline = new osis.CfnPipeline(
+  openSearchStack,
+  "OpenSearchIntegrationPipeline",
+  {
+    maxUnits: 4,
+    minUnits: 1,
+    pipelineConfigurationBody: openSearchTemplate,
+    pipelineName: "dynamodb-integration-2",
+    logPublishingOptions: {
+      isLoggingEnabled: true,
+      cloudWatchLogDestination: {
+        logGroup: logGroup.logGroupName,
+      },
+    },
+  }
+);
+
+```
+
+After deploying the resources, you can test the data ingestion process by adding an item to the `Todo` table. However, before doing that, let's verify that the pipeline has been set up correctly.
+
+In the AWS console, navigate to OpenSearch and then to the pipelines section. You should find your configured pipeline and review its settings to ensure they match your expectations:
+
+![A screenshot displaying the OpenSearch OSIS pipeline created under the DynamoDB integrations section](/images/gen2/opensearch-integration/OpenSearch_pipeline.png)
+
+You can also check this in the DynamoDB console by going to the Integrations section of the tables.
+
+![A screenshot displaying the OpenSearch OSIS pipeline created within the 'Ingestion -> Pipelines' section of the OpenSearch Console.](/images/gen2/opensearch-integration/OpenSearch_DynamoDB_integration.png)
+
+## Step 4: Expose new queries on OpenSearch
+
+### Step 4a:Add OpenSearch Datasource to backend
+
+First, Add the OpenSearch data source to the data backend.
+
+``` ts title="amplify/backend.ts"
+// Add OpenSearch data source 
+const osDataSource = backend.data.addOpenSearchDataSource(
+  "osDataSource",
+  openSearchDomain
+);
+```
+### Step 4b: Create Resolver and attach to query
+
+Let's create the search resolver. Create a new file named `amplify/data/searchBlogResolver.ts` and paste the following code. For additional details please refer to [Amazon OpenSearch Service Resolvers](https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-elasticsearch-resolvers-js.html)
+
+```ts title="amplify/data/searchBlogResolver.ts"
```

Comment:
is the file extension supposed to be js?

Reply:
Also, should the name be `searchTodoResolver` instead?